core/templates/pages/about-page/about-page.component.spec.ts

```diff
@@ -16,161 +16,208 @@
  * @fileoverview Unit tests for the about page.
  */
 
+import { ComponentFixture, fakeAsync, TestBed} from '@angular/core/testing';
+import { EventEmitter, NO_ERRORS_SCHEMA, Pipe, PipeTransform }
+  from '@angular/core';
+
+import { AboutPageComponent } from './about-page.component';
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { TranslateService } from 'services/translate.service';
+import { UtilsService } from 'services/utils.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
 import { WindowRef } from 'services/contextual/window-ref.service';
 
-require('pages/about-page/about-page.component.ts');
+@Pipe({name: 'translate'})
+class MockTranslatePipe implements PipeTransform {
+  transform(value: string, ...args):string {
+    return value;
+  }
+}
+
+class MockTranslateService {
+  code = 'es';
+  use(lang: string): string {
+    this.code = lang;
+    return this.code;
+  }
+}
+
+class MockI18nLanguageCodeService {
+  codeChange = new EventEmitter<string>();
+  getCurrentI18nLanguageCode() {
+    return 'en';
+  }
+
+  onI18nLanguageCodeChange() {
+    return this.codeChange;
+  }
+}
+
+// Mocking window object here because changing location.href causes the
```

Comment:
As mentioned above, try to mock minimally, here and elsewhere.

Reply:
Yes, this has been mocked minimally. I think no changes are required here.

Reply:
Can you use the existing windowRef and spyOn specific functions instead? That would be more minimal.

Reply:
I see, ok, thanks! I'm not against that a priori, though I wonder if we can do anything to avoid possible skew. Whatever convention we decide on, it might be worth making this a standard thing to follow (see above comment about wiki page).

Reply:
Will add this to wiki page